common.go

```diff
@@ -131,6 +131,12 @@ func (r *router) route(routes routeSlice, input net.HardwareAddr, src, dst net.I
 			}
 			if candSpec < curSpec {
 				continue
+			} else if candSpec == curSpec {
+				// it must use `<=` instead `<` because routes are sort by priority,
+				// we should treat earlier rtInfo more specific
+				if mostSpecificRt.Priority <= rt.Priority {
```

Comment:
This does flipe the selection logic for non-linux systems where priority is always 0.

previously the logic was just: `candSpec < curSpec`, so if a subsequent route is found with `candSpec == curSpec`, that route was selected over the previous one.
with this change, the equivalent-scoped subsequent matched route will be `continue`d over.

It does pass the limited test cases we have in CI,
but probably we should confirm the sorting logic on mac and windows in practice don't end up doing something unexpected because of this change.

Reply:
Encapsulate the route rule comparing logic into `IsMoreSpec` and keep it the same as before. 

See how to optimize this part later.  Is it feasible?